
* revert change from 1.43.2.4 to 1.43.2.5 
  it expects kld_isloaded() and kld_load() in libutil

Index: src/sbin/mdconfig/mdconfig.c
===================================================================
--- src.orig/sbin/mdconfig/mdconfig.c
+++ src/sbin/mdconfig/mdconfig.c
@@ -41,6 +41,7 @@
 static int md_find(char *, const char *);
 static int md_query(char *name);
 static int md_list(char *units, int opt);
+static void mdmaybeload(void);
 static char *geom_config_get(struct gconf *g, char *name);
 static void md_prthumanval(char *length);
 
@@ -253,9 +254,7 @@
 	}
 	mdio.md_version = MDIOVERSION;
 
-	if (!kld_isloaded("g_md") && kld_load("geom_md") == -1)
-		err(1, "failed to load geom_md module");
-
+	mdmaybeload();
 	fd = open("/dev/" MDCTL_NAME, O_RDWR, 0);
 	if (fd < 0)
 		err(1, "open(/dev/%s)", MDCTL_NAME);
@@ -454,3 +453,21 @@
 {
 	return (md_list(name, OPT_UNIT));
 }
+
+void
+mdmaybeload(void)
+{
+	char name1[64], name2[64];
+
+	snprintf(name1, sizeof(name1), "g_%s", MD_NAME);
+	snprintf(name2, sizeof(name2), "geom_%s", MD_NAME);
+	if (modfind(name1) == -1) {
+		/* Not present in kernel, try loading it. */
+		if (kldload(name2) == -1 || modfind(name1) == -1) {
+			if (errno != EEXIST) {
+				errx(EXIT_FAILURE,
+				    "%s module not available!", name2);
+			}
+		}
+	}
+}
